plugins/data-management/src/HierarchicalTrackSelectorWidget/components/faceted/FacetFilters.tsx

```diff
@@ -104,11 +15,24 @@ export default function FacetFilters({
   width: number
 }) {
   const facets = columns.slice(1)
-  const uniqs = facets.map(() => new Map<string, number>())
-  for (const row of rows) {
-    for (const [index, column] of facets.entries()) {
-      const elt = uniqs[index]
-      const key = `${row[column.field] || ''}`
+  const uniqs = new Map(
+    facets.map(f => [f.field, new Map<string, number>()] as const),
+  )
+  const filterKeys = Object.keys(filters)
+  const facetKeys = facets.map(f => f.field)
```

Comment:
this is the code that "stages the facets". it first adds any filters, in order, to a set